Azure.MgmtSdk.Analyzers/ModelNameSuffix.cs

```diff
@@ -12,43 +12,39 @@ namespace Azure.MgmtSdk.Analyzers
     /// Analyzer to check type name suffixes. There are some suffixed we don't recommend to use, like `Result`, `Response`...
     /// </summary>
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public class ModelNameSuffix : DiagnosticAnalyzer
+    public class ModelNameSuffixAnalyzer : DiagnosticAnalyzer
     {
-        public const string DiagnosticId = "AZM0010";
+        public const string DiagnosticIdBase = "AZM0010";
```

Comment:
`DiagnosticsId` should not be base. It should be bound to specific analyzer.

Reply:
We should have a base class which is abstract, then each concrete sub-class handle the certain cases. We should be cautious to inherit a concrete class.

Reply:
I suggest for specific informations like `DiagnosticDescriptor`, put into concrete class. `override` is better for methods.